mlflow/pytorch/__init__.py

```diff
@@ -240,6 +240,194 @@ def forward(self, x):
     )
 
 
+def log_state_dict(
```

Comment:
@harupy @chauhang 

`log_model` and `log_state_dict` uses `Model.log` to log the model inside mlflow. I have retained all the parameters which were there in `log_model` since there is no much code change. It would help us in some of the scenarios like, logging extra artifacts or registering the model when user logs the model as state dict. 

If you feel, we can reduce the argument and keep only few, please let me know.